#!/bin/sh /usr/share/dpatch/dpatch-run
## 211-magic-update-os2rexx.dpatch by Werner Fink <werner@suse.de>
##
## DP: Workaround regex DoS (CVE-2007-2026)

@DPATCH@

diff -Naur file-4.21.orig/magic/Magdir/msdos file-4.21/magic/Magdir/msdos
--- file-4.21.orig/magic/Magdir/msdos	2007-06-01 18:32:10.000000000 +0000
+++ file-4.21/magic/Magdir/msdos	2007-06-01 18:35:45.000000000 +0000
@@ -18,8 +18,10 @@
 100	regex/c =^[\ \t]{0,10}say\ ['"]	     OS/2 REXX batch file text
 
 
-100 regex/c =^\\s*call\\s+rxfuncadd.*sysloadfu OS/2 REXX batch file text
-100 regex/c =^\\s*say\ ['"] OS/2 REXX batch file text
+#100 regex/c =^\\s*call\\s+rxfuncadd.*sysloadfu OS/2 REXX batch file text
+#100 regex/c =^\\s*say\ ['"] OS/2 REXX batch file text
+100  regex/c =^\\s{0,255}call\\s{1,99}rxfuncadd	OS/2 REXX batch file text
+100  regex/c =^\\s{0,255}say\ ['"]		OS/2 REXX batch file text
 
 0	leshort		0x14c	MS Windows COFF Intel 80386 object file
 #>4	ledate		x	stamp %s
